nipap/nipap/authlib.py

```diff
@@ -327,17 +335,45 @@ def authenticate(self):
 
         # auth succeeded
         self.authenticated_as = self.username
-        self._authenticated = True
         self.trusted = False
         self.readonly = False
 
         try:
-            res = self._ldap_conn.search_s(self._ldap_basedn, ldap.SCOPE_SUBTREE, 'uid=' + self.username, ['cn'])
+            res = self._ldap_conn.search_s(self._ldap_basedn, ldap.SCOPE_SUBTREE, self._ldap_search.format(ldap.dn.escape_dn_chars(self.username)), ['cn','memberOf'])
```

Comment:
Not sure I follow, that's exactly what is intended, if both ro_group and rw_group is empty, no group member security is used, and everyone will always get read/write access no matter if memberOf is present or not. 

When I was testing I did not get NO_SUCH_ATTRIBUTE from the search_s if the memberOf attribute was missing, however I do get KeyError on 'if self._ldap_ro_group in res[0][1]['memberOf']:'

But sure, we can do a separate search for memberOf only if at least one of ro_group or rw_group is non empty.